From dbc8aefcb2abf0330a24cfbe6a75521393ed63cf Mon Sep 17 00:00:00 2001
Message-Id: <dbc8aefcb2abf0330a24cfbe6a75521393ed63cf.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Daniel Leung <daniel.leung@intel.com>
Date: Fri, 7 Mar 2014 17:20:32 -0800
Subject: [PATCH 040/312] REVERTME: HACK: drm/i915: Force HW reset/modeset at
 DPMS ON

Some LCD panels go blank after a DPMS off/on cycle. This hack
forces a reset so that we can at least use the device without
reboot.

Change-Id: I64c4469b4e17621fa74f87feaf47f635e6d4e4a3
Signed-off-by: Daniel Leung <daniel.leung@intel.com>
Reviewed-on: https://android.intel.com/170167
Reviewed-by: Boie, Andrew P <andrew.p.boie@intel.com>
Tested-by: Boie, Andrew P <andrew.p.boie@intel.com>
Reviewed-by: Englund, Hakan <hakan.englund@intel.com>
Reviewed-by: Gumbel, Matthew K <matthew.k.gumbel@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index c89f6a6..16ba224 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -5331,6 +5331,8 @@ static void intel_connector_check_state(struct intel_connector *connector)
  * consider. */
 void intel_connector_dpms(struct drm_connector *connector, int mode)
 {
+	struct drm_device *dev = connector->dev;
+
 	/* All the simple cases only support two dpms states. */
 	if (mode != DRM_MODE_DPMS_ON)
 		mode = DRM_MODE_DPMS_OFF;
@@ -5340,6 +5342,9 @@ void intel_connector_dpms(struct drm_connector *connector, int mode)
 
 	connector->dpms = mode;
 
+	if (mode == DRM_MODE_DPMS_ON)
+		intel_modeset_setup_hw_state(dev, true);
+
 	/* Only need to change hw state when actually enabled */
 	if (connector->encoder)
 		intel_encoder_dpms(to_intel_encoder(connector->encoder), mode);
-- 
1.7.9.5

